sys/include/net/ng_ipv6/addr.h

```diff
@@ -0,0 +1,507 @@
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+#define NG_IPV6_ADDR_ALL_NODES_IF_LOCAL {{ 0xff, 0x01, 0, 0, 0, 0, 0, 0, \
+                                           0, 0, 0, 0, 0, 0, 0, 1 }}
+
+/**
+ * @brief   Static initializer for the link-local all nodes multicast IPv6
+ *          address (ff02::1)
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+#define NG_IPV6_ADDR_ALL_NODES_LINK_LOCAL {{ 0xff, 0x02, 0, 0, 0, 0, 0, 0, \
+                                             0, 0, 0, 0, 0, 0, 0, 1 }}
+
+/**
+ * @brief   Static initializer for the interface-local all routers multicast IPv6
+ *          address (ff01::2)
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+#define NG_IPV6_ADDR_ALL_ROUTERS_IF_LOCAL {{ 0xff, 0x01, 0, 0, 0, 0, 0, 0, \
+                                             0, 0, 0, 0, 0, 0, 0, 2 }}
+
+/**
+ * @brief   Static initializer for the link-local all routers multicast IPv6
+ *          address (ff02::2)
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+#define NG_IPV6_ADDR_ALL_ROUTERS_LINK_LOCAL {{ 0xff, 0x02, 0, 0, 0, 0, 0, 0, \
+                                               0, 0, 0, 0, 0, 0, 0, 2 }}
+
+
+/**
+ * @brief   Static initializer for the site-local all routers multicast IPv6
+ *          address (ff05::2)
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+#define NG_IPV6_ADDR_ALL_ROUTERS_SITE_LOCAL {{ 0xff, 0x05, 0, 0, 0, 0, 0, 0, \
+                                               0, 0, 0, 0, 0, 0, 0, 2 }}
+
+/**
+ * @brief   Values for the flag field in multicast addresses.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+typedef enum {
+    /**
+     * @brief   The address is transient, i.e. not well-known, permanantly
+     *          assigned address by IANA.
+     */
+    NG_IPV6_ADDR_MCAST_FLAG_TRANSIENT = 0x01,
+
+    /**
+     * @brief   The address is based on a network prefix
+     *
+     * @see <a href="http://tools.ietf.org/html/rfc3306#section-4">
+     *          RFC 3306, section 4
+     *      </a>
+     */
+    NG_IPV6_ADDR_MCAST_FLAG_PREFIX_BASED = 0x02,
+
+    /**
+     * @brief   The address embeds the address on the rendezvous point
+     *
+     * @see <a href="http://tools.ietf.org/html/rfc3956#section-3">
+     *          RFC 3956, section 3
+     *      </a>
+     */
+    NG_IPV6_ADDR_MCAST_FLAG_EMBED_ON_RP = 0x04,
+} ng_ipv6_addr_mcast_flag_t;
+
+/**
+ * @brief   Values for the scope field in multicast addresses.
+ *
+ * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
+ *          RFC 4291, section 2.7
+ *      </a>
+ */
+typedef enum {
```

Comment:
There is also an `-fshort-enums` in gcc and clang.
On Feb 20, 2015 9:59 AM, "Oleg Hahm" <notifications@github.com> wrote:

> In sys/include/net/ng_ipv6/addr.h
> <https://github.com/RIOT-OS/RIOT/pull/2433#discussion_r25056422>:
>
> > +     *
> > +     * @see <a href="http://tools.ietf.org/html/rfc3956#section-3">
> > +     *          RFC 3956, section 3
> > +     *      </a>
> > +     */
> > +    NG_IPV6_ADDR_MCAST_FLAG_EMBED_ON_RP = 0x04,
> > +} ng_ipv6_addr_mcast_flag_t;
> > +
> > +/**
> > + * @brief   Values for the scope field in multicast addresses.
> > + *
> > + * @see <a href="http://tools.ietf.org/html/rfc4291#section-2.7">
> > + *          RFC 4291, section 2.7
> > + *      </a>
> > + */
> > +typedef enum {
>
> I thought that was only for packed structs. @thomaseichinger
> <https://github.com/thomaseichinger>, do you recall?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/RIOT-OS/RIOT/pull/2433/files#r25056422>.
>